### DISCUSSION TITLE
[v4.0.0] Benchmark performance analysis

### DISCUSSION THREAD

**Discussion by prescience-data** [colinhacks/zod#1038] [Ideas]
Just noticed the comparisons at https://github.com/moltar/typescript-runtime-type-benchmarks 

Are there any obvious "low hanging fruit" type architecture/implementation wins inspired by [valita](https://github.com/badrap/valita) that could be put on a roadmap for `v4.0.0`? 

<img width="662" alt="image" src="https://user-images.githubusercontent.com/65471523/159386990-69a2d6f4-5d73-4007-85bd-bade94b81933.png">

**Comment by zgayjjf** [comment]
This is shockingly slow

**Comment by divmgl** [comment]
How's this look now with the latest version of v3?
